classes/class-gf-field-hpsach.php

```diff
@@ -0,0 +1,416 @@
+     * @return array
+     */
+    function get_form_editor_field_settings() {
+        return array(
+            'force_ssl_field_setting',
+            'error_message_setting',
+            'label_setting',
+            'admin_label_setting',
+            'rules_setting',
+            'css_class_setting',
+        );
+    }
+
+    /**
+     * @return array
+     */
+    public function get_form_editor_button() {
+        return array(); // this button is conditionally added in the form detail page
+    }
+
+    /**
+     * @param array|string $value
+     * @param array        $form
+     */
+    public function validate($value, $form) {
+        $account_name = rgpost(GF_Field_HPSach::HPS_ACH_CHECK_HOLDER_FIELD_NAME);
+        $account_number = rgpost(GF_Field_HPSach::HPS_ACH_ACCOUNT_FIELD_NAME);
+        $routing_number = rgpost(GF_Field_HPSach::HPS_ACH_ROUTING_FIELD_NAME);
+        $account_type = rgpost(GF_Field_HPSach::HPS_ACH_TYPE_FIELD_NAME);
+        $check_type = rgpost(GF_Field_HPSach::HPS_ACH_CHECK_FIELD_NAME);
+
+        if ($this->isRequired && (empty($account_name) || empty($account_number) || empty($routing_number) || empty($account_type))) {
+            $this->failed_validation = true;
+            $this->validation_message = empty($this->errorMessage) ? esc_html__('Please enter your account information.', 'gravityforms') : $this->errorMessage;
+        }
+        elseif (!empty($card_number)) {
+            $card_type = GFCommon::get_card_type($card_number);
+
+            if (empty($security_code)) {
+                $this->failed_validation = true;
+                $this->validation_message = esc_html__("Please enter your card's security code.", 'gravityforms');
+            }
+            elseif (!$card_type) {
+                $this->failed_validation = true;
+                $this->validation_message = esc_html__('Invalid credit card number.', 'gravityforms');
+            }
+            elseif (!$this->is_card_supported($card_type['slug'])) {
+                $this->failed_validation = true;
+                $this->validation_message = $card_type['name'] . ' ' . esc_html__('is not supported. Please enter one of the supported credit cards.', 'gravityforms');
+            }
+        }
+    }
+
+    /**
+     * @param $card_slug
+     *
+     * @return bool
+     */
+    public function is_card_supported($card_slug) {
+        $supported_cards = $this->creditCards;
+        $default_cards = array('amex', 'discover', 'mastercard', 'visa');
+
+        if (!empty($supported_cards) && in_array($card_slug, $supported_cards)) {
+            return true;
+        }
+        elseif (empty($supported_cards) && in_array($card_slug, $default_cards)) {
+            return true;
+        }
+
+        return false;
+
+    }
+
+
+    /**
+     * @param array  $form
+     * @param string $value
+     * @param null   $entry
+     *
+     * @return string
+     */
+    public function get_field_input($form, $value = '', $entry = null) {
```

Comment:
This method could use some cleanup. There are credit card related items. Unnecessary items should be removed.